src/main/java/org/embulk/input/zendesk/services/ZendeskNormalServices.java

```diff
@@ -111,6 +121,29 @@ private void importDataForIncremental(final ZendeskInputPlugin.PluginTask task,
                         continue;
                     }
 
+                    // Contain some records  that later than end_time. Checked and don't add.
+                    // Because the api already sorted by updated_at or timestamp for ticket_events, we just need to break no need to check further.
+                    if (resultEndTime > endTime) {
+                        long checkedTime = 0;
+                        if (recordJsonNode.has(ZendeskConstants.Field.UPDATED_AT) && !recordJsonNode.get(ZendeskConstants.Field.UPDATED_AT).isNull()) {
+                            checkedTime = ZendeskDateUtils.isoToEpochSecond(recordJsonNode.get(ZendeskConstants.Field.UPDATED_AT).textValue());
+                        }
+
+                        // ticket events is updated by system not user's action so it only has timestamp field
+                        if (task.getTarget().equals(Target.TICKET_EVENTS) && recordJsonNode.has("timestamp") && !recordJsonNode.get("timestamp").isNull()) {
+                            checkedTime = recordJsonNode.get("timestamp").asLong();
+                        }
+
+                        // scores (or response) is only store rated_at time
+                        if (task.getTarget().equals(Target.SCORES) && recordJsonNode.has("rated_at") && !recordJsonNode.get("rated_at").isNull()) {
+                            checkedTime = ZendeskDateUtils.isoToEpochSecond(recordJsonNode.get("rated_at").textValue());
+                        }
+
+                        if (checkedTime > endTime) {
```

Comment:
There is a scenario where data is still correct but the purpose of limiting API does not work out.

If `start_time` is 2010 and `end_time` is 2012, responded data in 1st request has 1000 records, the number of record has been saved here: https://github.com/treasure-data/embulk-input-zendesk/blob/bee72fc08c9383b4e3e3c186e554853995e1dcb4/src/main/java/org/embulk/input/zendesk/services/ZendeskNormalServices.java#L113-L115, in these 1000 records, there are 500 records satisfy the condition from `start_time` to `end_time`, other 500 are from 2012 to 2014 which don't satisfy. Due to the check of `checkedTime > endTime`, the loop of iterator will be broken, then other request will be made because: https://github.com/treasure-data/embulk-input-zendesk/blob/bee72fc08c9383b4e3e3c186e554853995e1dcb4/src/main/java/org/embulk/input/zendesk/services/ZendeskNormalServices.java#L173-L175 is still satisfy.

Reply:
fixed